relayer/chains/cosmos/query.go

```diff
@@ -887,11 +887,24 @@ func (cc *CosmosProvider) QueryChannels(ctx context.Context) ([]*chantypes.Ident
 	qc := chantypes.NewQueryClient(cc)
 	p := DefaultPageRequest()
 	chans := []*chantypes.IdentifiedChannel{}
+	res := &chantypes.QueryChannelsResponse{}
 
 	for {
-		res, err := qc.Channels(ctx, &chantypes.QueryChannelsRequest{
-			Pagination: p,
-		})
+		err := func() error {
```

Comment:
this should probably be it's own function shared by this and `QueryChannelsPaginated`

Reply:
great call! i'll make that change